### ISSUE TITLE
Please add support for FreeBSD

### ISSUE THREAD

**Issue by yzgyyang** [microsoft/vscode-remote-release#727]
`Can't connect to azure: unreachable or not Linux x86_64 (FreeBSD amd64 )`

Since the Electron support is premature on FreeBSD, running vscode in my laptop and connect to my FreeBSD server seems to be a great solution. I'm looking forward to the support on FreeBSD platforms.

**Comment by mike-pt** [comment]
Would love to have this too, seems like the most work would be around vscode server but considering the deps you list for Linux it seems entirely possible.

**Comment by Qint** [comment]
me too.

**Comment by Qint** [comment]
> The remote server architecture is not supported

----------

The server's OS is FreeBSD.

**Comment by morganwdavis** [comment]
```
[20:21:58.866] > Running remote connection script
> 
[20:21:58.875] > Unsupported architecture: FreeBSD amd64   
> a993f545-b325-411d-80a3-23ece1e3d1b3##27##
> 
[20:21:58.875] Received install output: a993f545-b325-411d-80a3-23ece1e3d1b3##27##
[20:21:58.875] Unsupported architecture
[20:21:58.875] The remote server architecture is not supported
```
What would happen if the architecture check was simply ignored/skipped? Would it just work? I can't imagine what kind of esoteric commands being issued over a standard SSH connection would be so architecture-dependent that Linux would work but not FreeBSD.  If it's just a matter of using full paths to executables or environment variables, I'm sure we FreeBSD users could create a compatible environment.

**Comment by mike-pt** [comment]
I actually tried to work around that it does work if u fake it (meaning it passes this check) but it will then try to install things using what would be expected in linux, some would need to be convert to FreeBSD, but more importantly we need vscode-server built for FreeBSD

**Comment by morganwdavis** [comment]
Yes. I saw that it creates a .vscode-server directory.  Client software that silently installs stuff in the user's account is unexpectedly hacky and unfriendly. (I just got a horrible flashback to those ghastly server-side FrontPage web extensions provided by Microsoft 20 years ago!)

I think Microsoft should specify the server environment they require and leave it up to the developer community to create their own compatibility packages for their OS.  Then it can be cleanly built and maintained by the best folks who know best, managed using standard package management tools (e.g., pkg under FreeBSD).  The user's home directory would then have a set of symlinks or similar mechanism for providing the expected support interfaces.

This can't come soon enough for me. I'm finding that vscode's git features over the Internet to a remote Samba server is excruciatingly slow -- taking a full minute to do a simple commit. (Any tips/workarounds are appreciated!)

**Comment by Follpvosten** [comment]
> more importantly we need vscode-server built for FreeBSD

I'm just guessing, but this shouldn't be too hard, since the server is probably built with TS/JS, right? If the source is available, I'd like to help with this.

**Comment by mike-pt** [comment]
Thats the issue though the source is not available AFAIK, Microsoft would need to built it and release it, so it can be downloaded on first setup, this is the critical part here.

I didn't find any reference to the source so far.

**Comment by mike-pt** [comment]
I suppose if someone has the Linux zip we could look into it and see if freebsd Linux emulation works.

**Comment by qqkookie** [comment]
Same problem. I can contribute to solve this problem. What is main problem that prevents VS Code  from supporting FreeBSD? Is it fault of FreeBSD ssh server or Open SSH or VS Code remote SSH extension problem?

**Comment by mike-pt** [comment]
The main problem is what I described, the vscode server component is not built/distributed  for FreeBSD, since its also not open source only Microsoft can do it.

Alternately perhaps we could look into using Linux emulation and then patch the extension side to use the Linux version and do whatever is necessary to get it to install on freebsd, the issue is not ssh or the extension it self (that probably easy to handle).

**Comment by davidchisnall** [comment]
This is currently the issue blocking me from using VS Code.  Please let me know if there's anything I can do to help make this work.

**Comment by Rob4226** [comment]
+1 to please add support for FreeBSD!!!   I am sure there are many others who would use it also.  It can't be too hard to adapt it to FreeBSD given that it is largely based on the SSH protocol.  Thanks!!!

**Comment by krowvin** [comment]
Agreed, please add FreeBSD support!

**Comment by geunyul** [comment]
+1 Agreed, please add FreeBSD support!

**Comment by LongJohnSilver7** [comment]
+1 Agreed, please add FreeBSD support!

**Comment by resnerac03** [comment]
Please add FreeBSD support ASAP. I can't work on my favorite text editor coz of this.

**Comment by arymkus** [comment]
Please add FreeBSD support.

**Comment by Follpvosten** [comment]
It's great that we have some spiking activity here, the issue might get some attention, but I'm not sure how constructive it is.

I'd like to add something: Please point us to the server source code so *we* can add FreeBSD support! I'd be willing to help, and I'm sure others would be, too.

**Comment by mike-pt** [comment]
@Follpvosten +1 I've pointed out the same before, they would need to either release that themselves or open source it so someone can try to build it. AFAIK this is not open sourced (the server code I mean).

**Comment by dghiringhelli** [comment]
Please add FreeBSD support!

**Comment by dpetrov** [comment]
+1 for FreeBSD support.

**Comment by luchoman08** [comment]
+1 for FreeBSD support.

**Comment by xcdr** [comment]
+1, nice to have FreeBSD support.

**Comment by RocketSquirrel** [comment]
Concur on adding support for FreeBSD. I'm currently relegated to using Eclipse, and while it is sufficient, it's very heavy-handed compared to VSCode.

**Comment by freebobbyshmurda** [comment]
+1

**Comment by tkanjar** [comment]
+1

**Comment by hansinator** [comment]
+1

**Comment by cdsim** [comment]
+1, it would be great if you could support FreeBSD !

**Comment by guilhermedrud** [comment]
+1

**Comment by cwlin0416** [comment]
+1

**Comment by cookys** [comment]
+1, it would be great to have FreeBSD support.

**Comment by jimying** [comment]
+1

**Comment by Rob4226** [comment]
+1

**Comment by chickenwing** [comment]
+1

**Comment by netops2devops** [comment]
+1

**Comment by do-pudding** [comment]
+1

**Comment by eogloblin** [comment]
+1

**Comment by fedqx** [comment]
+1

**Comment by ohmantics** [comment]
+1

**Comment by dyeske** [comment]
+1

**Comment by cleoold** [comment]
+1, nice to have FreeBSD support. (in 2020)

**Comment by alexseitsinger** [comment]
+1

**Comment by jimying** [comment]
Please add Remote SSH support, freebsd is always as server, remotely develop will convenient

**Comment by nkev** [comment]
+1

**Comment by mishrapuru** [comment]
+1

**Comment by thenicknash** [comment]
+1

**Comment by duartejbrito** [comment]
+1

**Comment by sh1dow3r** [comment]
+1 for FreeBSD support

**Comment by shellqiqi** [comment]
+1 for FreeBSD support

**Comment by sean0921** [comment]
**+1**

**Comment by isarantis** [comment]
+1

**Comment by Rabenda** [comment]
+1

**Comment by zeon-ua** [comment]
+1

**Comment by danniehansen** [comment]
+1

**Comment by zatarain** [comment]
+1

**Comment by leyang95** [comment]
+1

**Comment by kentippens** [comment]
+1

**Comment by mrschofield** [comment]
+1

**Comment by pjbates** [comment]
+1

**Comment by bobmagicii** [comment]
+1

**Comment by DavMig** [comment]
Hello, is there an expected date for this ? thanks

**Comment by zatarain** [comment]
I don't think they will even consider it, unfortunately :(

**Comment by nhooyr** [comment]
code-server works on FreeBSD! It's VS Code but running in the browser instead of electron.

https://github.com/cdr/code-server#quick-install

**Comment by mike-pt** [comment]
Thanks and that sure helps, but remote vscode allows you to connect to several machines via ssh from the same place, now that there port in freebsd seems to work fine and even code server works as you posted this shouldn't be hard to implement I imagine, its just a matter of getting prioritized, its one of the most voted features here so there is "demand" for it :) 🤞

**Comment by bobmagicii** [comment]
i will literally delete and remake a few cloud instances i made this past month once this gets settled. i had them all set up on freebsd it never even crossed my mind this would not work. so at the moment they are ubuntu.

**Comment by zatarain** [comment]
Does someone knows how can I actually clone the code of vscode-remote? So, I can try to add something by my self and then try to contribute.

**Comment by nhooyr** [comment]
> Does someone knows how can I actually clone the code of vscode-remote? So, I can try to add something by my self and then try to contribute.

It's unfortunately closed source.

**Comment by bobmagicii** [comment]
> 
> 
> > Does someone knows how can I actually clone the code of vscode-remote? So, I can try to add something by my self and then try to contribute.
> 
> It's unfortunately closed source.

which is really interesting considering microsoft's recent pushes about things.

**Comment by jonasnordlund** [comment]
This was an unfortunate surprise and makes it difficult to work with NearlyFreeSpeech hosting. :-(

**Comment by davidone** [comment]
Any update on this?

**Comment by noxmoe** [comment]
+1

**Comment by zatarain** [comment]
> > Does someone knows how can I actually clone the code of vscode-remote? So, I can try to add something by my self and then try to contribute.
> 
> It's unfortunately closed source.

We are hopeless 😢

**Comment by mike-pt** [comment]
I must admit I'm a bit disappointed at Microsoft since this is so high voted here, and since vscode works on FreeBSD now there's really no excuse I think .... Its fine if they want to keep this part closed source, but considering how many people have requested this a least a word on if this planed or not would be nice..

**Comment by zatarain** [comment]
I'm too, but to be fair is understandable. I mean, they started to offer support for Linux, I think they never said will start to be open for all *nix world, which is really sad unfortunately 😞

**Comment by amit-nath** [comment]
Please support for FreeBSD

**Comment by jimying** [comment]
+1 Please add support

**Comment by maisen20** [comment]
+1

**Comment by madisonblue** [comment]
+1

**Comment by caio-vinicius** [comment]
+1 for FreeBSD

**Comment by yaazkal** [comment]
It'll be pretty great to support FreeBSD

+1 for this feature

**Comment by pousterlus** [comment]
Hi everyone,

Adding my 2 cents to ask for FreeBSD support.

**Comment by xpader** [comment]
Needs freebsd support too!

**Comment by zouyonghao** [comment]
Any update?

**Comment by mike-pt** [comment]
Sadly nope... It remains one of the most requests features though: https://github.com/Microsoft/vscode-remote-release/issues?q=is%3Aopen+is%3Aissue+label%3Afeature-request+sort%3Areactions-%2B1-desc


I was truly hoping this would be in the roadmap at least since there's even a port for vscode itself, and code-server works in a FreeBSD host (https://github.com/cdr/code-server, this has been my workaround until eventually, Microsoft picks this up)

**Comment by szporwolik** [comment]
+1 for FreeBSD

**Comment by yaazkal** [comment]
> I was truly hoping this would be in the roadmap at least since there's even a port for vscode itself, and code-server works in a FreeBSD host (https://github.com/cdr/code-server, this has been my workaround until eventually, Microsoft picks this up)

@mike-pt I've tried code-server on a jail some days ago. It just shows "Cannot reconnect. Please reload the window." over and over again. So no way to work with it. I'll appreciate if you can share your configuration so I can see if I'm able to run it on my jail. I'm Using FreeBSD 12.1p10

**Comment by mike-pt** [comment]
I haven't tried in a Jail yet, but I had issues in the host using yarn and npm worked (or the other way around ). I later had issue using search and replace and was able to find help on their GitHub. You might get faster help if you just open an issue there.

**Comment by tprelog** [comment]
another +1 for FreeBSD support

**Comment by natalka1122** [comment]
+1 for FreeBSD.

**Comment by ykla** [comment]
+1 for FreeBSD.

**Comment by jimying** [comment]
[code-server](https://github.com/cdr/code-server), should be work, but not official supply freebsd build binary, not convenient as    
linux

**Comment by 36roentgen** [comment]
+1 for FreeBSD.

**Comment by fahadshery** [comment]
+1 for FreeBSD.
I am using MacOS to connect to FreeBSD Jail.
I am can ssh into FreeBSD jail without any issues.  However, when connecting via vscode I get this error:
```code
[17:58:32.289] Log Level: 2
[17:58:32.291] remote-ssh@0.61.0
[17:58:32.291] darwin x64
[17:58:32.293] SSH Resolver called for "ssh-remote+ha_jail", attempt 1
[17:58:32.293] "remote.SSH.useLocalServer": true
[17:58:32.294] "remote.SSH.sshPath": undefined
[17:58:32.294] "remote.SSH.sshConfigurationFile": undefined
[17:58:32.294] "remote.SSH.useFlock": true
[17:58:32.294] "remote.SSH.lockfilesInTmp": false
[17:58:32.294] "remote.SSH.localServerDownload": auto
[17:58:32.295] "remote.SSH.remoteServerListenOnSocket": false
[17:58:32.295] "remote.SSH.showLoginTerminal": false
[17:58:32.295] "remote.SSH.maxReconnectionAttempts": 9007199254740991
[17:58:32.295] SSH Resolver called for host: ha_jail
[17:58:32.295] Setting up SSH remote "ha_jail"
[17:58:32.299] Acquiring local install lock: /var/folders/2s/jzqgfm0x5qddggpk0__qv2dh0000gn/T/vscode-remote-ssh-homeassistant_jail-install.lock
[17:58:32.313] Looking for existing server data file at /Users/Fahad/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-homeassistant_jail-e5a624b788d92b8d34d1392e4c4d9789406efe8f-0.61.0/data.json
[17:58:32.314] Using commit id "e5a624b788d92b8d34d1392e4c4d9789406efe8f" and quality "stable" for server
[17:58:32.318] Install and start server if needed
[17:58:32.325] Checking ssh with "ssh -V"
[17:58:32.334] > OpenSSH_8.1p1, LibreSSL 2.7.3

[17:58:32.337] askpass server listening on /var/folders/2s/jzqgfm0x5qddggpk0__qv2dh0000gn/T/vscode-ssh-askpass-6552e9c10564a38e60cfabc27fba2de6f5cd09e2.sock
[17:58:32.338] Spawning local server with {"ipcHandlePath":"/var/folders/2s/jzqgfm0x5qddggpk0__qv2dh0000gn/T/vscode-ssh-askpass-d4118b1aa570228a1c8100f53270c17224915774.sock","sshCommand":"ssh","sshArgs":["-v","-T","-D","55805","-o","ConnectTimeout=60","homeassistant_jail"],"dataFilePath":"/Users/Fahad/Library/Application Support/Code/User/globalStorage/ms-vscode-remote.remote-ssh/vscode-ssh-host-ha_jail-e5a624b788d92b8d34d1392e4c4d9789406efe8f-0.61.0/data.json"}
[17:58:32.338] Local server env: {"DISPLAY":"1","ELECTRON_RUN_AS_NODE":"1","SSH_ASKPASS":"/Users/Fahad/.vscode/extensions/ms-vscode-remote.remote-ssh-0.61.0/out/local-server/askpass.sh","VSCODE_SSH_ASKPASS_NODE":"/Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper (Renderer).app/Contents/MacOS/Code Helper (Renderer)","VSCODE_SSH_ASKPASS_MAIN":"/Users/Fahad/.vscode/extensions/ms-vscode-remote.remote-ssh-0.61.0/out/askpass-main.js","VSCODE_SSH_ASKPASS_HANDLE":"/var/folders/2s/jzqgfm0x5qddggpk0__qv2dh0000gn/T/vscode-ssh-askpass-6552e9c10564a38e60cfabc27fba2de6f5cd09e2.sock"}
[17:58:32.341] Spawned 78295
[17:58:32.459] > local-server> Spawned ssh: 78296
[17:58:32.463] stderr> OpenSSH_8.1p1, LibreSSL 2.7.3
[17:58:32.568] stderr> debug1: Server host key: ecdsa-sha2-nistp256 SHA256:xpSFxeKEjm+pzDWop+2GRoWp+alaHqOSKxL31k0D3tE
[17:58:32.675] stderr> Authenticated to XXX.XXX.X.XXX ([XXX.XXX.X.XXX]:22).
[17:58:32.708] > Having trouble using fetch through a firewall? Try setting the environment
> variable FTP_PASSIVE_MODE to yes, and see fetch(3) for more details.
[17:58:32.709] > ready: 6c8849c640ca
[17:58:32.716] > FreeBSD 12.2-RC3 FreeBSD 12.2-RC3 7c4ec6ff02c(HEAD) TRUENAS 
[17:59:34.342] Terminating local server
[17:59:34.350] Resolver error: Error: Connecting with SSH timed out
	at Function.Timeout (/Users/Fahad/.vscode/extensions/ms-vscode-remote.remote-ssh-0.61.0/out/extension.js:1:82451)
	at Timeout._onTimeout (/Users/Fahad/.vscode/extensions/ms-vscode-remote.remote-ssh-0.61.0/out/extension.js:1:351807)
	at listOnTimeout (internal/timers.js:531:17)
	at processTimers (internal/timers.js:475:7)
[17:59:34.352] Local server exit: 15
[17:59:34.357] ------
```

thanks

**Comment by csfreak** [comment]
+1

**Comment by grahamperrin** [comment]
@tagattie with [editors/vscode](https://www.freshports.org/editors/vscode/) at `1.46.1`, please: when might we have an update?

https://code.visualstudio.com/updates/ `1.52.1` at the time of writing. 

Thank you

**Comment by jimboca** [comment]
+1 Please support FreeBSD!

**Comment by grahamperrin** [comment]
@jimboca please don't use comments to up-vote. Instead, please use the up-vote feature of GitHub.

**Comment by zheng8501** [comment]
+1 please support FreeBSD

**Comment by jimying** [comment]
+1

**Comment by dfr** [comment]
+1

**Comment by elovelan** [comment]
This comment section reflects poorly on the FreeBSD community (i.e. we aren't willing to read [the rules](https://github.com/Microsoft/vscode-remote-release/blob/main/CONTRIBUTING.md#look-for-an-existing-issue)):

> If you find your issue already exists, make relevant comments and add your reaction. Use a reaction in place of a "+1" comment:
> 
> - 👍 - upvote
> - 👎 - downvote

Also, all of these extraneous comments make it really hard to find helpful comments!

@Rob4226, @krowvin, @geunyul, @n0x32schneider, @arymkus, @dghiringhelli, @dpetrov, @luchoman08, @xcdr, @RocketSquirrel, @tkanjar, @hansinator, @cdsim, @guilhermedrud, @cwlin0416, @cookys, @jimying (4 comments???), @Rob4226, @chickenwing, @netops2devops, @eogloblin, @fedqx, @ohmantics, @dyeske, @cleoold, @alexseitsinger, @nkev, @thenicknash, @duartejbrito, @sh1dow3r, @shellqiqi, @sean0921, @isarantis, @Rabenda, @zeon-ua, @danniehansen, @zatarain, @leyang95, @kentippens, @mrschofield, @pjbates, @bobmagicii, @amit-nath, @maisen20, @madisonblue, @caio-vinicius, @yaazkal, @pousterlus, @xpader, @szporwolik, @tprelog, @natalka1122, @ykla, @36roentgen, @fahadshery, @csfreak, @zheng8501, @dfr please remove your comment(s) and 👍 the first comment if you haven't already.

Comments to help other users get [cdr/code-server](https://github.com/cdr/code-server) or similar solutions working (especially in a jail) would be more than welcome; especially since someone like @Chuxel may see them and push a similar solution in MS's private repo for this feature!

**Comment by Rob4226** [comment]
@elovelan 🙄

**Comment by davidchisnall** [comment]
@Rob4226, would you be willing to submit a port containing that so that folks can install code-server as a package?

**Comment by Rob4226** [comment]
> @Rob4226, would you be willing to submit a port containing that so that folks can install code-server as a package?

@davidchisnall Sure, that's a good idea. I'll work on one.

**Comment by xpader** [comment]
Code-server is not full function like vscode-remote, alots of extensions can not install on code-server, Can microsoft consider to supporting FreeBSD?
There seems to be no movement of this.

**Comment by blindone0** [comment]
Please make it.

**Comment by Linkoid01** [comment]
Very much interested in this.

**Comment by da-tai** [comment]
Learning Freebsd, would love this added

**Comment by xpader** [comment]
I hope fleet can running on freebsd now.

**Comment by maisen20** [comment]
Really a pity that this is still not supported. So we can't upload via SFTP plugin over sftp but only over ftp and with this unencrypted connection I just don't have a secure feeling.

**Comment by han-fastolfe** [comment]
+1. I use FreeBSD servers everywhere.

**Comment by frank716** [comment]
+1

**Comment by DoLooP** [comment]
+1

**Comment by ohmantics** [comment]
I tried again a few years later and not even the Remote-SSH plugin can target FreeBSD, which remains closed source. It's odd that the Azure team has gone out of their way to court FreeBSD users for their cloud, sponsoring FreeBSD conferences and such, but the tools folks at Microsoft seem to have their own priorities.

**Comment by bobmagicii** [comment]
> folks at Microsoft seem to have their own priorities.

there is ok, there is another bug on the main repo where all people want is a button to clear the problems pane and its been years.

**Comment by davidchisnall** [comment]
> I tried again a few years later and not even the Remote-SSH plugin can target FreeBSD, which remains closed source. It's odd that the Azure team has gone out of their way to court FreeBSD users for their cloud, sponsoring FreeBSD conferences and such, but the tools folks at Microsoft seem to have their own priorities.

Summarising the discussion from above (not speaking for the maintainers of this project):

There is no strong objection from the VS Code team to supporting FreeBSD but the way that the system is currently designed makes this very difficult due to external dependencies.  Currently, VS Code does not provide plugins with an different interface between running things on the host and the remote system.  Adding an abstraction layer to do that and porting existing plugins across would be a huge engineering effort.  I'd imagine that you'd want to start by providing remote file access proxies and then gradually enlightening plugins to use explicit communication and that's a difficult problem (for example, should I run `clangd` or some other LSP server locally and require copying the entire source tree across the network to index, or should I run it remotely and require a big program on a system that might be a low-end embedded target?).  To avoid this, the remote extension installs VS Code on the remote system and runs it headless, proxing commands, so any plugin that you have on your local system can be installed remotely and work there.

This means that the requirement for the remote extension to work on a given platform is the same as the requirement for VS Code to work on that platform.  This brings us to the next set of problems:

VS Code uses Electron.  Electron will support any platform that their dependencies run on but they are mostly a wrapper around Chromium and Node.js.  Node.js works fine on FreeBSD, Chromium does not.  Chromium is a Google project and they support only platforms that Google uses internally or sells products on (Android, iOS, macOS, Windows, desktop Linux, and Fuchsia).  To make this worse, the Chromium project has explicitly refused to take patches to support FreeBSD, even with external volunteers maintaining them and providing CI infrastructure, so there is no obvious path to supporting FreeBSD in Chromium.

Neither the Electron team nor the VS Code team is willing to take on the effort of maintaining a fork of Chromium that supports FreeBSD.  If you want VS Code to support FreeBSD then you should encourage Google to accept the FreeBSD patches for Chromium.  Until they decide to, or an antitrust regulator forces them to, all downstream projects are blocked. 

This issue has [the second-highest number of thumb-up reactions of any issue on this repo](https://github.com/microsoft/vscode-remote-release/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc), so it's likely that it would receive some attention if it were unblocked upstream.

**Comment by dyeske** [comment]
David Chisnall, so you are saying the blocker here is google and chromium.  Can you share the patches that need to go upstream?  I have maintained over 40 FreeBSD ports in the last 20 years.  If we need to pull in some chromium bits and patch then as a vscode dependency we can probably do that.  It wouldn't be the first time.  Sometimes short solutions are required to get the long term solution completed.  Feel free to contact me directly.

**Comment by xpader** [comment]
Look around files in .vscode-server folder on server, looks like there's no any file exists about chromium binary, only node and js files, base on this, looks like vscode-remote doesn't need Electron's features by itself, may be this is more of an architectural problem?
I didn't view any vscode-remote codes, it's just my humble opinion.

**Comment by mateuszkwiatkowski** [comment]
So it seems that linuxulator is good enough to make remote ssh play with FreeBSD.
I prepared a gist with little howto: https://gist.github.com/mateuszkwiatkowski/ce486d692b4cb18afc2c8c68dcfe8602

I have working both remote code editing and terminal. One user reported that terminal doesn't work for him on 13.0 (The terminal process failed to launch: A native exception occurred during launch (forkpty(3) failed.).).

**Comment by xpader** [comment]
@mateuszkwiatkowski Linuxulator is not full compatible with vscode-remote-ssh, I have tried with Ubuntu 20.04 on FreeBSD 13.0, vscode-remote worked, but some error happened in logs, file monitor notify with inotify not work, so vscode can't detect file change in explorer, some socket options not adapter, and other. when close vscode, remote process keep high cpu usage in linuxulator long times.
And create linuxulator is not easy.

**Comment by mateuszkwiatkowski** [comment]
@xpader I don't agreee that linuxulator setup is not easy - its documentation is great and after initial setup it just works. While I'm aware that inotify is not supported by default in linuxulator there's a way to get it working with [libinotify-kqueue](https://github.com/libinotify-kqueue/libinotify-kqueue) and [linux-libbsd](https://github.com/wulf7/linux-libbsd) shims. The latter is currently broken on CURRENT but with some community effort it can be fixed and provide even better experience with vscode-remote-ssh. Having that said I'm currently using remote-ssh and it works great for me, vscodes notices file modifications even without inotify.

**Comment by BasixKOR** [comment]
For those of you who rather want to hack around instead of waiting, I found an workaround here: https://blog.david-reid.com/vs-code/

**Comment by unknowndevQwQ** [comment]
> For those of you who instead want to hack around instead of waiting, you may try code-server instead: https://blog.david-reid.com/vs-code/

@BasixKOR 
Will there be any problems if I use freebsd's npm?
One more thing, the install.sh of coder/code-server now fully supports freebsd(with npm)

**Comment by BasixKOR** [comment]
Not a support issue for code-server. Ask anywhere else.

**Comment by openingnow** [comment]
Can vscode from pkg(which is, in fact, vscodium) work as remote extension host? If then, we can send patch to [remote-oss extension](https://github.com/xaberus/vscode-remote-oss) to support FreeBSD and use vscodium.(or make vscode extension with vscodium host?)

**Comment by jjantura** [comment]
I can confirm LSP (haskell-language-server) working.

**Comment by shaojing1779** [comment]
> Unsupported platform: FreeBSD
> 940a177439e8: start
> exitCode==35==
> listeningOn====
> osReleaseId==freebsd==
> arch====
> vscodeArch====
> bitness====
> tmpDir==/tmp==
> platform====
> unpackResult====
> didLocalDownload==0==
> downloadTime====
> installTime====
> extInstallTime====
> serverStartTime====
> 940a177439e8: end


openbsd & freebsd cant work

**Comment by sueszli** [comment]
Unfortunately still no support - although it has already been ported: https://www.freshports.org/editors/vscode/

**Comment by grahamperrin** [comment]
> it has already been ported: [freshports.org/editors/vscode](https://www.freshports.org/editors/vscode/)

If you check all commentary above (including the sixty-something hidden comments), you might find that what's requested here is somewhat different from what's ported. 

---

A gentle hint, to other commenters: if your comment was (essentially) no more than a `+1`, please consider deleting. Allow substantial commentary to be more easily found. Thanks.

**Comment by vedranmiletic** [comment]
With VS Code 1.86, the [support for CentOS 7 is no longer available](https://code.visualstudio.com/updates/v1_86#_linux-minimum-requirements-update) and the `linux_base-c7`/`linux-c7-*` packages, which are the only Linux packages provided in FreeBSD Ports, are no longer helpful to get VS Code remote running. There is no better time than now to consider officially supporting FreeBSD.

**Comment by masteradhoc** [comment]
@roblourens @chrmarti any plans to support this? +1 from my side

**Comment by DavMig** [comment]
Hello, +1 for this as FreeBSD is widely used.

**Comment by Malix-Labs** [comment]
> FreeBSD is widely used

I am also using FreeBSD on some of my servers but let's not lie

**Comment by coreybrett** [comment]
I would switch my Python/Flask web apps over to run on top of FreeBSD if VSC was compatible with it.

**Comment by bobmagicii** [comment]
i legit have a pile of servers on ubuntu purely because this didn't work years ago when i needed it to. >_>

**Comment by vedranmiletic** [comment]
FWIW, it is possible to use RL9 Linuxulator to run VS Code Remote, I wrote up a small howto: https://group.miletic.net/en/blog/2024-06-14-how-to-run-visual-studio-vs-code-remote-over-ssh-on-freebsd-13-and-14/

**Comment by jirpok** [comment]
> FWIW, it is possible to use RL9 Linuxulator to run VS Code Remote, I wrote up a small howto: https://group.miletic.net/en/blog/2024-06-14-how-to-run-visual-studio-vs-code-remote-over-ssh-on-freebsd-13-and-14/

Don't forget to remove `Hostname` and `User` lines from `~/.ssh/config` like I did, because otherwise it doesn't work.

**Comment by qureshifiroj88** [comment]
[20:21:58.866] > Running remote connection script
> 
[20:21:58.875] > Unsupported architecture: FreeBSD amd64   
> a993f545-b325-411d-80a3-23ece1e3d1b3##27##
> 
[20:21:58.875] Received install output: a993f545-b325-411d-80a3-23ece1e3d1b3##27##
[20:21:58.875] Unsupported architecture
[20:21:58.875] The remote server architecture is not supported

**Comment by jpcastberg** [comment]
Noting the extension that allows for limited freebsd support is broken as of the latest vscode release. https://github.com/SchoofsKelvin/vscode-sshfs/issues/452

**Comment by xpader** [comment]
There is no meaning for VSCode to support FreeBSD now, Vibe Coding we don't need VSCode now.
